addons/web/static/src/search/search_model.js

```diff
@@ -257,7 +257,7 @@ export function getDomainTreeDescription(
         const values = Array.isArray(value) ? value : [value];
         const formatValue = (value) => {
             if (typeof value === "string") {
-                return value;
+                return formatValueFromFieldType(value, fieldDef);
```

Comment:
add 2 tests:
- one in domain_selector_tests (with readonly = true) (set initial domain with a datetime and a date -> check display)
- one in search_bar_tests (edit a filter -> add a datetime and a date -> save -> check facet)

Reply:
There is already one test like that in search_bar_tests. I've added 2 tests in domain_selector_tests